### PR TITLE
Update for Moodle 3.0

### PR DESCRIPTION
**Pull request by timhodson**
Adds in new event logging classes to move away from deprecated functions (`add_to_log()`).
Adds in a number of other minor deprecation fixes for Moodle 3.

### CODE REVIEW

File: 2.x-activity-module/mod/aspirelists/classes/event/aspirelists_launch.php [hunk 1]
```diff
@@ -0,0 +1,35 @@
+<?php
+namespace aspirelists\event;
+
+defined('MOODLE_INTERNAL' || die);
+
+class aspire_lists_launch extends \core\event\base {
```
Comment (lordtatty) [inline 2.x-activity-module/mod/aspirelists/classes/event/aspirelists_launch.php]: Can we get a docblock to explain what this class is for.

File: 2.x-activity-module/mod/aspirelists/launch.php [hunk 2]
```diff
@@ -4,6 +4,9 @@
 require_once($CFG->dirroot.'/mod/aspirelists/lib.php');
 require_once($CFG->dirroot.'/mod/lti/lib.php');
 require_once($CFG->dirroot.'/mod/lti/locallib.php');
+require_once($CFG->dirroot.'/mod/aspirelists/classes/event/aspirelists_launch.php');
+
+//use aspirelists\event\aspire_lists_launch;
```
Comment (lordtatty) [inline 2.x-activity-module/mod/aspirelists/launch.php]: If this isn't needed, can we remove this line?

**Comment by lordtatty** [comment]
Is this going to break in older versions of moodle?  Our travis build is only going back as far as 2.8 at the moment - is it worth adding more back to 2.5?  Looking at the moodle github it looks like ``\core\event\base`` was introduced in 2.6, so I'm guessing everything from there on should work.

File: 2.x-activity-module/mod/aspirelists/launch.php [hunk 3]
```diff
@@ -15,8 +18,17 @@
 $cm = get_coursemodule_from_id('aspirelists', $id, 0, false, MUST_EXIST);
 $list = $DB->get_record('aspirelists', array('id' => $cm->instance), '*', MUST_EXIST);
 $course = $DB->get_record('course', array('id' => $cm->course), '*', MUST_EXIST);
+$context = context_module::instance($cm->id);
 
-add_to_log($course->id, "aspirelists", "launch", "launch.php?id=$cm->id", "$list->id");
```
Comment (lordtatty) [inline 2.x-activity-module/mod/aspirelists/launch.php]: Which versions of moodle are going to lose the ability to log with this line removed?

File: 2.x-activity-module/mod/aspirelists/classes/event/course_module_viewed.php [hunk 4]
```diff
@@ -0,0 +1,27 @@
+<?php
+namespace aspirelists\event;
+defined('MOODLE_INTERNAL' || die);
+
+class course_module_viewed extends \core\event\course_module_viewed {
```
Comment (lordtatty) [inline 2.x-activity-module/mod/aspirelists/classes/event/course_module_viewed.php]: Can we get a dockblock to explain the purpose of this class

File: 2.x-activity-module/mod/aspirelists/view.php [hunk 5]
```diff
@@ -39,6 +39,9 @@
 require_once($CFG->dirroot.'/mod/lti/lib.php');
 require_once($CFG->dirroot.'/mod/lti/locallib.php');
 require_once($CFG->dirroot . '/lib/completionlib.php');
+require_once($CFG->dirroot.'/mod/aspirelists/classes/event/course_module_viewed.php');
+
+//use aspirelists\classes\event\course_module_viewed;
```
Comment (lordtatty) [inline 2.x-activity-module/mod/aspirelists/view.php]: Can we remove this use if it's not necessary

File: tools/development/install_plugins.sc [hunk 6]
```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
```
Comment (lordtatty) [inline tools/development/install_plugins.sc]: Did you mean to check this in?

**Comment by lordtatty** [comment]
It looks good, though I think we need to figure out what the implications are for moodle version support here.

Also some of those methods could do with some docblocks.

File: tools/development/install_plugins.sc (hunk 6 above)
Comment (timhodson) [inline tools/development/install_plugins.sc]: Yes.

**Comment by timhodson** [comment]
So I have added in builds for 2.6 and 2.7 too.
Also addressed PR comments.

File: 2.x-activity-module/mod/aspirelists/launch.php (hunk 3 above)
Comment (timhodson) [inline 2.x-activity-module/mod/aspirelists/launch.php]: added version specific defensive check round this.

**Comment by lordtatty** [comment]
👍

**Pull request merged**